### ISSUE TITLE
Cluttering warnings

### ISSUE THREAD

**Issue by effigies** [poldracklab/fmriprep#1981]
Somewhere we're importing `nipype.interfaces.dipy` and getting:

```
200213-15:05:46,646 nipype.interface INFO:
         We advise you to upgrade DIPY version. This upgrade will open access to more function
200213-15:05:46,649 nipype.interface INFO:
         We advise you to upgrade DIPY version. This upgrade will open access to more function
200213-15:05:46,650 nipype.interface INFO:
         We advise you to upgrade DIPY version. This upgrade will open access to more models
```

This is inappropriate for fMRIPrep. I don't know if this is one of our packages loading too much and hitting these or if Nipype promiscuously loads too many interfaces.

I'll update this thread with any others I find while running.

#1512 redux.

**Comment by oesteban** [comment]
I think the problem is both. In MRIQC I'll soon try with quite a drastic approach only for non-dev releases (setting `os.environ['PYTHONWARNINGS'] = 'ignore'`).

Lately, the `FutureWarning` from sklearn are even more annoying and hard to redirect. In this case, those are hit via nilearn which loads the deprecated joblib from sklearn.

Because of multiprocessing, warnings of the subprocesses are not filtered. To propagate the warning configurations, you can only modify the environment or modify the initialization callback of the Processor Pool inside nypipe to set up all the filters.

Alternatively, not eagerly preloading everything on `__init__.py` of both `niworkflows` (for this particular FutureWarning) and `nipype` (for the case of dipy) would also help a lot (and perhaps scratch some memory load off of the workers).

**Comment by mgxd** [comment]
closed via #2080
